har/har.go

```diff
@@ -371,17 +373,30 @@ func (l *Logger) RecordRequest(id string, req *http.Request) error {
 		return err
 	}
 
-	l.mu.Lock()
-	defer l.mu.Unlock()
-
-	l.entries[id] = &Entry{
+	entry := &Entry{
 		ID:              id,
 		StartedDateTime: time.Now().UTC(),
 		Request:         hreq,
 		Cache:           &Cache{},
 		Timings:         &Timings{},
 	}
 
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	if _, exists := l.entries[id]; exists {
+		return fmt.Errorf("Duplicate request ID: %s", id)
+	}
+	l.entries[id] = entry
+	if l.tail == nil {
+		l.tail = entry
+		l.tail.next = entry
```

Comment:
return out of here, and remove the else - the stuff in the else block can be pulled into theprimary flow